tests/functional/olp-cpp-sdk-authentication/AuthenticationCommonTestFixture.cpp

```diff
@@ -98,6 +100,11 @@ AuthenticationCommonTestFixture::AcceptTerms(
 AuthenticationTestUtils::DeleteUserResponse
 AuthenticationCommonTestFixture::DeleteUser(
     const std::string& user_bearer_token) {
+  // Before deleting user we must wait some time before server finishes
+  // SignUp operation. In case we delete user just after SignUp
+  // ( which is unreal use case ) server will send us an error.
+  std::this_thread::sleep_for(kDeleteUserWait);
```

Comment:
Would it make sense replacing the sleep with something more reliable? E.g. looking up if the user successfully signed up.

Reply:
we do a signup check later but when server is not completed deleting user it sends us http status 404 which is not the right thing to do because we may miss broken url later.